src/NUnitEngine/nunit.engine.core/ITestAgent.cs

```diff
@@ -24,18 +24,53 @@
 namespace NUnit.Engine
 {
     /// <summary>
-    /// The ITestAgent interface is implemented by remote test agents.
+    /// Defines the current communication protocol between the engine and the agent. This is an implementation detail
+    /// which is in the process of changing as the dependency on .NET Framework's remoting is removed.
     /// </summary>
     public interface ITestAgent
     {
         /// <summary>
         /// Stops the agent, releasing any resources
         /// </summary>
-        void Stop();
+        void ShutDown();
 
         /// <summary>
-        ///  Creates a test runner
+        /// Loads a package which will be used for all subsequent calls to the other methods (except
+        /// <see cref="ShutDown"/>). This method must be called at least once before calling methods that operate on a
+        /// test package. It may be called any number of times after those methods.
         /// </summary>
-        ITestEngineRunner CreateRunner(TestPackage package);
+        TestEngineResult Load(TestPackage package);
```

Comment:
Reading this, I feel the need to read all the code so as to see if all the methods of `ITestEngineResult` are present. They are, so why not just inherit from that interface?